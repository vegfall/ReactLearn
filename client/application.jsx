import React, {useState} from "react";

export function ListMovies() {
    return <div>
        <h1>List movies!</h1>
        <ul>
            <li>Movie 1</li>
            <li>Movie 2</li>
        </ul>
    </div>
}

export function NewMovieForm({onAddMovie}) {
    const[title, setTitle] = useState("");
    const[year, setYear] = useState("");
    const[plot, setPlot] = useState("");

    function handleSubmit() {
        onAddMovie({title, year, plot});
    }

    return <form onSubmit={handleSubmit}>
        <h1>Add new movie</h1>
        <div>
            Title: <input type="text" data-testid="title" value={title} onChange={e => setTitle(e.target.value)}/>
        </div>
        <div>
            Year: <input type="text" data-testid="year" value={year} onChange={e => setTitle(e.target.value)}/>
        </div>
        <div>
            Plot:
            <div><textarea data-testid="plot" value={plot} onChange={e => setPlot(e.target.value)}/></div>
        </div>
    </form>
}