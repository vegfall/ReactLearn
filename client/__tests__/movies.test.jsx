import React from "react";
import renderer, { act } from "react-test-renderer";

import {ListMovies} from "../application";

describe("movies application", () => {
    it('should show movie list', () => {
        expect(6 * 9).toEqual(54);
    });

    it("renders correctly", () => {
        const component = renderer.create(<h1>Hello world!</h1>);
        expect(component).toMatchSnapshot();
    });

    it("matches snapshot", async () => {
        let component;

        await act(async () => {
            component = renderer.create(<ListMovies/>);
        });

        expect(component).toMatchSnapshot();
    });
});