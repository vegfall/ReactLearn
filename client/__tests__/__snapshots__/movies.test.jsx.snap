// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies application matches snapshot 1`] = `
<div>
  <h1>
    List movies!
  </h1>
  <ul>
    <li>
      Movie 1
    </li>
    <li>
      Movie 2
    </li>
  </ul>
</div>
`;

exports[`movies application renders correctly 1`] = `
<h1>
  Hello world!
</h1>
`;

exports[`movies application show new movie form 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Add new movie
  </h1>
  <div>
    Title: 
    <input
      data-testid="title"
      type="text"
    />
  </div>
  <div>
    Year: 
    <input
      data-testid="year"
      type="text"
    />
  </div>
  <div>
    Plot:
    <div>
      <textarea
        data-testid="plot"
      />
    </div>
  </div>
</form>
`;
